test/basic.html

```diff
@@ -359,6 +366,20 @@
         server.restore();
       });
 
+      test('fires iron-form-invalid if it can\'t submit', function(done) {
+        var form = f.querySelector('#mixed-invalid');
+        form.addEventListener('iron-form-invalid', function(event) {
+          expect(form.validate()).to.be.equal(false);
+          done();
+        });
+        // Wait one tick for observeNodes.
+        Polymer.Base.async(function() {
+          expect(form.validate()).to.be.equal(false);
+          form.submit();
+          server.respond();
+        }, 1);
```

Comment:
I hear ya, just trying to understand things better myself :) Appreciate the work you have put into this!